pkgs/ffigen/lib/src/visitor/fix_overridden_methods.dart

```diff
@@ -41,4 +45,43 @@ class FixOverriddenMethodsVisitation extends Visitation {
       }
     }
   }
+
+  void _fixMethodsVsProperties(ObjCInterface node) {
+    // In ObjC, supertypes and subtypes can have a method that's an ordinary
+    // method in some classes of the heirarchy, and a property in others. This
+    // isn't allowed in Dart, so we change all such conflicts to properties.
+    // This change could cause more conflicts in the heirarchy, so first we walk
+    // up to find the root of the subtree that has this method, then we walk
+    // down the subtreee to change all conflicting methods to properties.
+    for (final method in node.methods) {
+      final (root, rootMethod) = _findRootWithMethod(node, method);
+      if (method.isProperty == rootMethod.isProperty) continue;
+      _convertAllSubtreeMethodsToProperties(root, rootMethod);
+    }
+  }
+
+  (ObjCInterface, ObjCMethod) _findRootWithMethod(
+      ObjCInterface node, ObjCMethod method) {
+    var root = node;
+    var rootMethod = method;
+    for (ObjCInterface? t = node; t != null; t = t.superType) {
+      final tMethod = t.getMethod(method.originalName);
+      if (tMethod != null) {
+        root = t;
+        rootMethod = tMethod;
+      }
+    }
+    return (root, rootMethod);
+  }
+
+  void _convertAllSubtreeMethodsToProperties(
```

Comment:
What about the case where some code is already generated with methods in another package and we want to interop with that code instead of regenerating the parent classes? In that case we would want to convert everything to methods instead.

Reply:
Hmmm. There's also the case where two imported packages disagree about whether a method is a getter or an ordinary method, in which case there's no valid choice. I think I'll probably need an option to control whether getters are generated eventually. I think these edge cases are pretty unlikely though, so I'll defer that for now.